### PR TITLE
Merge generics into main repository

### PR DESCRIPTION
**Pull request by RLKRo**
changes:
- add generics
- cover all by testing except venv and build
- fix style of examples and tests

### CODE REVIEW

**Comment by pseusys** [comment]
Documentation inspection here.

In order to make documentation for this branch buildable by `make doc` command some changes should be made:
1. In order to be excluded from gallery, `examples/generics/example_utils.py` should start from underscore (be renamed to `_example_utils.py`).
2. Examples should not contain `input()` invocations run during building. If some examples are meant to be executed both in interactive and automatic mode, an `-a` command line flag is provided during sphinx gallery execution and can be provided during testing as well (see `docs/source/conf.py:101`). You can see how this argument can be used during example running in pipeline examples, for instance.
3. Example root should contain a `README.rst` file with an `RST` header (see `examples/pipeline/README.rst`).
4. Each python script in example root should start with a docstring containing an `RST` header (file name should be a good choice). You can look at pipeline examples for inspiration, for instance.

That's it for now!

**Review by ruthenian8 — APPROVED** [review]

File: examples/pipeline/4_groups_and_conditions_full.py [hunk 1]
```diff
@@ -97,7 +115,7 @@ def runtime_info_printing_service(_, __, info: ServiceRuntimeInfo):
         [
             simple_service,  # This simple service will be named `simple_service_0`
             simple_service,  # This simple service will be named `simple_service_1`
-        ],  # Despite this is the only unnamed service group in the root service group, it will be named `service_group_0`
+        ],  # Despite this is the unnamed service group in the root service group, it will be named `service_group_0`
```
[the remark this answers is not in the document]
Comment (kudep) [reply examples/pipeline/4_groups_and_conditions_full.py:118]: We need reduce a length of the line.

File: makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ SHELL = /bin/bash
 
 PYTHON = python3
 VENV_PATH = venv
-VERSIONING_FILES = setup.py makefile docs/source/conf.py dff/__init__.py
+VERSIONING_FILES = setup.py makefile docs/source/conf.py .__init__.py
```
[the remark this answers is not in the document]
Comment (kudep) [reply makefile:5]: fix #19

File: examples/pipeline/4_groups_and_conditions_full.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RLKRo) [reply examples/pipeline/4_groups_and_conditions_full.py:118]: Not by completely removing the point of the sentence. Might as well remove the sentence altogether.
Let the line be
"Despite this is the only unnamed service group in the root service group, it is named `service_group_0`".
That sentence is also 5 characters shorter but it keeps the purpose of the original sentence.

**Pull request merged**